assets/js/src/integrations/classic.js

```diff
@@ -60,14 +60,20 @@ export function classicTracking(
 		}
 		// Get product ID from data attribute (archive pages) or value (single product pages).
 		const productID = parseInt(
-			button[ 0 ].dataset.product_id || button[ 0 ].value
+			button?.[ 0 ]?.dataset.product_id || button?.[ 0 ].value
 		);
 
+		if ( isNaN( productID ) ) {
```

Comment:
Although `parseInt` won't return a type other than number or `NaN`, it would be better to avoid using `isNaN` when possible because it has some [confusing special-case behavior](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#description) that leads to semantics ambiguity.